spring-cloud-kubernetes-test-support/src/main/java/org/springframework/cloud/kubernetes/integration/tests/commons/native_client/Util.java

```diff
@@ -465,6 +473,22 @@ private void waitForDeploymentToBeDeleted(String deploymentName, String namespac
 		});
 	}
 
+	private void waitForDeploymentPodsToBeDeleted(Map<String, String> labels, String namespace) {
```

Comment:
Wait for the deployment pods to be deleted also. I had issues with some builds if I do not wait for this too, because two subsequents tests can step on each others toes when I search for them. If you need more verbose details, I can provide them, but it seems reasonable to wait for pods deletion also.